tree/tree.h

```diff
@@ -0,0 +1,47 @@
+#ifndef _TREE_H__ 
+#define _TREE_H__
+
+#include "splitter.h"
+
+#include <vector>
+
+namespace tree {
+
+/**
+ * Base class for all Decision Tree implementations.
+ */
+class Tree{
+
+  public:
+
+    /**
+     * \brief Standard destructor.
+     */
+    virtual ~Tree() { }
```

Comment:
The virtual destructor shouldn't have the body implementation. I guess just this would be enought:

`virtual ~Tree()`